clients/go/zbc/envWrapper.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zbc
+
+import (
+	"os"
+)
+
+var systemEnv = newEnvironmentWrapper()
+
+type environment interface {
+	getenv(variable string) string
+	setenv(variable, value string)
+	unsetenv(variable string)
+	lookupEnv(variable string) (string, bool)
+	environCopy() map[string]string
+	overwrite(environ map[string]string)
+}
+
+type envWrapper struct {
+	vars map[string]string
+}
+
+func (wrapper *envWrapper) setenv(variable, value string) {
+	wrapper.vars[variable] = value
+}
+
+func (wrapper *envWrapper) getenv(variable string) string {
+	value, _ := wrapper.lookupEnv(variable)
+	return value
+}
+
+func (wrapper *envWrapper) unsetenv(variable string) {
+	if _, present := wrapper.vars[variable]; present {
+		delete(wrapper.vars, variable)
+		_ = os.Unsetenv(variable)
+	}
+}
+
+func (wrapper *envWrapper) lookupEnv(variable string) (string, bool) {
+	if value, ok := wrapper.vars[variable]; ok {
+		return value, ok
+	}
+
+	return os.LookupEnv(variable)
+}
+
+func (wrapper *envWrapper) environCopy() map[string]string {
+	envDup := make(map[string]string, len(wrapper.vars))
+
+	for envVar, val := range wrapper.vars {
+		envDup[envVar] = val
+	}
+
+	return envDup
+}
+
+func (wrapper *envWrapper) overwrite(environ map[string]string) {
+	wrapper.vars = environ
```

Comment:
I would also copy it here to be more hygienic (i.e. the passed map cannot be modified to modify our current env)